src/cljs/bundled/lumo/closure.cljs

```diff
@@ -2255,83 +2255,69 @@
     (let [opts  (cond-> opts
                   (= (:verbose opts :not-found) :not-found)
                   (assoc :verbose true))
-          paths (map #(Paths/get (.toURI %)) (-paths source))
-          path  (first paths)
-          fs    (.getFileSystem path)
-          srvc  (.newWatchService fs)]
+          paths (if (seq? source)
+                  source (vector source))
+          throttling? (volatile! false)]
       (letfn [(buildf []
                 (try
-                  (let [start (System/nanoTime)]
+                  (let [start (system-time)]
                     (build source opts compiler-env)
                     (println "... done. Elapsed"
-                      (/ (unchecked-subtract (System/nanoTime) start) 1e9) "seconds")
-                    (flush))
+                      (-> (system-time) (- start) (/ 1e3) (.toFixed 2)) "seconds"))
                   (when-let [f (:watch-fn opts)]
                     (f))
-                  (catch Throwable e
+                  (catch js/Error e
                     (if-let [f (:watch-error-fn opts)]
                       (f e)
-                      (binding [*out* *err*]
-                        (println (Throwables/getStackTraceAsString e)))))))
-              (watch-all [^Path root]
-                (Files/walkFileTree root
-                  (reify
-                    FileVisitor
-                    (preVisitDirectory [_ dir _]
-                      (let [^Path dir dir]
-                        (. dir
-                          (register srvc
-                            (into-array [StandardWatchEventKinds/ENTRY_CREATE
-                                         StandardWatchEventKinds/ENTRY_DELETE
-                                         StandardWatchEventKinds/ENTRY_MODIFY])
-                            (into-array [SensitivityWatchEventModifier/HIGH]))))
-                      FileVisitResult/CONTINUE)
-                    (postVisitDirectory [_ dir exc]
-                      FileVisitResult/CONTINUE)
-                    (visitFile [_ file attrs]
-                      FileVisitResult/CONTINUE)
-                    (visitFileFailed [_ file exc]
-                      FileVisitResult/CONTINUE))))]
+                      (binding [*print-fn* *print-err-fn*]
+                        (println e))))))
+              (watch-all [root]
+                (.watch fs root #js {:recursive false}
+                        (fn [change fstr]
+                          (let [rel-path (.join path root fstr)]
```

Comment:
nit: can we use `path/join` and `fs/existsSync`?

Reply:
yes ofc, forgot to realize path and fs are imported as namespaces.